velox/core/CoreTypeSystem.h

```diff
@@ -192,6 +193,18 @@ class ArrayValReader : public ArrayValWriter<VAL> {
       append(std::move(val));
     }
   }
+
+  bool mayHaveNulls() const {
+    VELOX_NYI();
+  }
+
+  std::optional<VAL> operator[](size_t index) const {
```

Comment:
Those are needed since the DynamicExec environment type system still uses them, 
and the evalDynamic functions calls instantiate the functions for that path.

However, I would like to completely  the DynamicExec path and environment since its not used anywhere. 
and then we can remove those.